index.html

```diff
@@ -3549,28 +3548,37 @@ <h3><dfn>Writing or pushing content</dfn></h3>
         the <a>activated reader objects</a>, run the following sub-steps:
         <ol>
           <li>
-            If |reader|.<a>[[\Id]]</a> is [= dictionary member/present =]
-            and it is not equal to any |record|'s
-            <a href="#dom-ndefrecord-id">id</a> where |record|
-            is an element of |message|, [= iteration/continue =].
-          </li>
-          <li>
-            If |reader|.<a>[[\RecordType]]</a> is [= dictionary member/present =]
-            and it is not equal to any |record|'s
-            <a href="#dom-ndefrecord-recordtype">recordType</a> where |record|
-            is an element of |message|, [= iteration/continue =].
-          </li>
-          <li>
-            If |reader|.<a>[[\MediaType]]</a> is not `""` and
-            it is not equal to any |record|'s
-            <a href="#dom-ndefrecord-mediatype">mediaType</a> where |record| is
-            an element of |message|, [= iteration/continue =].
-          </li>
-          <li>
-            <a>Fire an event</a> named "`reading`" at |reader|
-            using <a>NDEFReadingEvent</a> with its <a>serialNumber</a> attribute
-            initialized to |serialNumber| and <a>message</a> attribute
-            initialized to |message|.
+            For each |record:NDEFRecord| in |message:NDEFMessage|,
+            <ol>
+              <li>
+                Let |matched| be `undefined`.
+              </li>
+              <li>
+                If |reader|.<a>[[\Id]]</a> is not `""` and is not equal to
+                |record|'s <a href="#dom-ndefrecord-id">id</a>, set |matched|
+                to `false`. If it is equal, set |matched| to `true`.
+              </li>
+              <li>
+                If |reader|.<a>[[\RecordType]]</a> is
+                [= dictionary member/present =] and it is not equal to
+                |record|'s <a href="#dom-ndefrecord-recordtype">recordType</a>,
+                set |matched| to `false`. If it is equal and if
+                |matched| is `undefined`, set |matched| to `true`.
+              </li>
+              <li>
+                If |reader|.<a>[[\MediaType]]</a> is not `""` and is not equal
+                to |record|'s <a href="#dom-ndefrecord-mediatype">mediaType</a>,
+                set |matched| to `false`. If it is equal and if |matched| is
+                `undefined`, set |matched| to `true`.
+              </li>
+              <li>
+                If |matched| is `true`, <a>fire an event</a> named "`reading`"
```

Comment:
Include this in `for each` will lead to multiple times triggering of `reading` event if multiple records in the message matched. Because we are always dispatching the matched message rather than matched record, this maybe another problem.

Reply:
The step says to abort the algorithm if a match is found.

Reply:
Which was bad as well, as it should continue with the next reader instead of aborting.

Reply:
Per spec:
> [Break ]((https://infra.spec.whatwg.org/#iteration-break))will skip over any remaining steps in an iteration, and skip over any remaining items as well, stopping the iteration.

"break" would be more appropriate.

Reply:
Yes. Changed.